libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Schemas/Actions/Microsoft.BeginDialog.schema

```diff
@@ -19,12 +19,19 @@
             ]
         },
         "dialog": {
-            "oneOf": [
+            "anyOf": [
```

Comment:
>anyOf [](start = 13, length = 5)

Should be oneOf.

Reply:
removed stringExpression. 
However, using oneOf key here. The shemaMergeTests will fail. Is there any other factor will cause the failure?

Reply:
I tried oneOf, however `TestDialogResourcesAreValidForSchema` test on the newly added test.dialog files would throw errors.

Reply:
Try it again with your new schema.  This is exactly what you have for AdaptiveDialog and either it should work both places or neither.

---
In reply to: [520477479](https://github.com/microsoft/botbuilder-dotnet/pull/4880#discussion_r520477479) [](ancestors = 520477479)

Reply:
@chrimc62 
I found the root cause of problem. 
The pattern key in "Microsoft.IDialog" is not effective. It should be move to the string definition in `Microsoft.IDialog.schema`.
```
{
                        "$kind": "Microsoft.IDialog",
                        "pattern": "^(?!(=)).*",
                        "title": "Dialog"
                    }
```

Otherwise, strings like `=settings.dialogID` would be valid both for "$kind": "Microsoft.IDialog" and "equalsExpression"  rules. I edited the schema files. Could you take a look?